java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1821_java.html

```diff
@@ -0,0 +1,5 @@
+<p>Nested <code>switch</code> structures are difficult to understand because you can easily confuse the cases of an inner <code>switch</code> as
+belonging to an outer statement. Therefore nested <code>switch</code> statements should be avoided.</p>
+<p>Specifically, you should structure your code to avoid the need for nested <code>switch</code> statements, but if you cannot, then consider moving
+the inner <code>switch</code> to another function.</p>
+
```

Comment:
This is not enough. We need a `Compliant` and `Noncompliant` example for java. To do so, create a subtask of [RSPEC-1821](https://jira.sonarsource.com/browse/RSPEC-1821) called "Java" and put some simple examples of code. Then regenerate the documentation.